### PR TITLE
fix DeleteCustomTheme

### PR DESCRIPTION
**Pull request by altynbek132**
found bug (did not test). 

decrement selected id if deleted id preceded it

### CODE REVIEW

**Comment by altynbek132** [comment]
I did not test because I have trouble with receiving Twitter API. Could you help me, please? What should I choose and write?

[Twitter Developers.pdf](https://github.com/robertodoering/harpy/files/7140361/Twitter.Developers.pdf)

File: lib/components/settings/theme_selection/bloc/theme_event.dart [hunk 1]
```diff
@@ -155,16 +155,27 @@ class DeleteCustomTheme extends ThemeEvent with HarpyLogger {
       final lightThemeId = app<ThemePreferences>().lightThemeId;
       final darkThemeId = app<ThemePreferences>().darkThemeId;
 
-      // reset selection when deleting a selected theme
-      if (themeId == lightThemeId || themeId == darkThemeId) {
+      if (themeId >= lightThemeId || themeId >= darkThemeId) {
```
Comment (robertodoering) [inline lib/components/settings/theme_selection/bloc/theme_event.dart:158]:
```suggestion
      if (themeId <= lightThemeId || themeId <= darkThemeId) {
```

File: lib/components/settings/theme_selection/bloc/theme_event.dart [hunk 2]
```diff
@@ -155,16 +155,27 @@ class DeleteCustomTheme extends ThemeEvent with HarpyLogger {
...
+          lightThemeId: _adjustedThemeId(lightThemeId),
+          darkThemeId: _adjustedThemeId(darkThemeId),
           saveSelection: true,
         ));
       }
     }
   }
+
+  // reset selection when deleting a selected theme
+  // or decrement if deleted id preceded selected id
```
Comment (robertodoering) [inline lib/components/settings/theme_selection/bloc/theme_event.dart:169]: Let's give this method proper dart doc, for example:

```suggestion
  /// Returns the new `themeId` of the currently selected theme.
```

File: lib/components/settings/theme_selection/bloc/theme_event.dart [hunk 3]
```diff
@@ -155,16 +155,27 @@ class DeleteCustomTheme extends ThemeEvent with HarpyLogger {
...
+  int? _adjustedThemeId(int currentThemeId) {
+    if (themeId == currentThemeId) {
+      return 0;
+    }
+    if (themeId >= currentThemeId) {
```
Comment (robertodoering) [inline lib/components/settings/theme_selection/bloc/theme_event.dart:174]:
```suggestion
    if (themeId <= currentThemeId) {
```

File: lib/components/settings/theme_selection/bloc/theme_event.dart [hunk 4]
```diff
@@ -155,16 +155,27 @@ class DeleteCustomTheme extends ThemeEvent with HarpyLogger {
...
+      return currentThemeId - 1;
+    }
+    return null;
+  }
```
Comment (robertodoering) [inline lib/components/settings/theme_selection/bloc/theme_event.dart:178]: Let's also move this method above `applyAsync`.
We always sort `applyAsync` last in events for better visibility (just like `build` is always sorted last in widgets)

**Review by robertodoering — COMMENTED** [review]
Thanks @altynbek132 and nice catch, this was indeed a bug.
We need to check whether the deleted theme's id is smaller than the currently selected id though.

**Comment by robertodoering** [comment]
Looks great, thanks!
We can ignore the failing build, since master is broken for the new flutter update right now

**Pull request merged**